lib/widgets/components/purchase/purchase_overlay.dart

```diff
@@ -37,35 +34,22 @@ class PurchaseOverlay {
             child: BlocListener<PurchaseCubit, PurchaseState>(
               listener: (context, state) async {
                 if (state is PurchaseCompleted) {
-                  hide(); //Removes this overlay
-                  // FIXME: Correct navigation
-                  Navigator.pop(context);
+                  final payment = state.payment;
+                  payment.productName =
+                      product.name; //TODO Receive this from backend
+                  Navigator.pop<Payment>(context, payment);
+
                   //TODO Consider if these calls should be moved elsewhere, e.g. inside the purchase cubit
                   final ticketCubit = sl.get<TicketsCubit>();
```

Comment:
These final lines of code should just be:

```dart
sl.get<TicketsCubit>().getTickets();
sl.get<ReceiptCubit>().fetchReceipts();
```

No need to await, as it doesn't do anything.

Reply:
In order to avoid a confusing situation in the future, if someone were to expand upon the method, I would definitely say that we need to await them. Even if it does not currently serve a purpose to do so. 

In #237 I did simplify the calls by not assigning the cubits to variables, as you suggest here. DO you want it as part of this PR, or are you fine with it being in the other one, considering I am moving where these calls happen anyway?

Reply:
Fine as a part of another PR